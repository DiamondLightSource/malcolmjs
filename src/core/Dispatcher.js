/*
 * React.js Starter Kit
 * Copyright (c) 2014 Konstantin Tarkus (@koistya), KriaSoft LLC.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import Flux from 'flux';
import PayloadSources from '../constants/PayloadSources';
import assign from 'react/lib/Object.assign';

/**
 * A singleton that operates as the central hub for application updates.
 * For more information visit https://facebook.github.io/flux/
 */
let Dispatcher = assign(new Flux.Dispatcher(), {

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the server.
   */
  handleServerAction(action) {
    var payload = {
      source: PayloadSources.SERVER_ACTION,
      action: action
    };
    this.dispatch(payload);
  },

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the view.
   */
  handleViewAction(action) {
    var payload = {
      source: PayloadSources.VIEW_ACTION,
      action: action
    };
    this.dispatch(payload);
  }

});

export default Dispatcher;
