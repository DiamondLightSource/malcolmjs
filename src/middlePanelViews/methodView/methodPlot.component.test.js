import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import MockCircularBuffer from '../../malcolm/reducer/attribute.reducer.mocks';

import MethodPlot from './methodPlot.component';

const mockTheme = {
  palette: {
    primary: { light: '#f0f0f0' },
    text: { primary: '#000' },
    background: { default: '#424242' },
  },
  alarmState: { warning: '#fff', error: '#ff0000', disconnected: '#0500ff' },
};

describe('MethodPlot', () => {
  let shallow;
  let testArchive;

  beforeEach(() => {
    shallow = createShallow({ dive: true });
    testArchive = {
      value: new MockCircularBuffer(5),
      timeStamp: new MockCircularBuffer(5),
      alarmState: new MockCircularBuffer(5),
    };
    testArchive.value.push({
      runParameters: { test1: 0 },
      returned: { test2: 1 },
    });
    testArchive.timeStamp.push({
      localRunTime: new Date(0),
      localReturnTime: new Date(4),
    });
    testArchive.alarmState.push(0);
    testArchive.value.push({
      runParameters: { test1: 0 },
      returned: { test2: 1 },
    });
    testArchive.timeStamp.push({
      localRunTime: new Date(2),
      localReturnTime: new Date(5),
    });
    testArchive.alarmState.push(1);
    testArchive.value.push({
      runParameters: { test1: 0 },
      returned: { test2: 1 },
    });
    testArchive.timeStamp.push({
      localRunTime: new Date(3),
      localReturnTime: new Date(6),
    });
    testArchive.alarmState.push(2);
  });

  it('renders correctly', () => {
    const wrapper = shallow(
      <MethodPlot
        attribute={testArchive}
        openPanels={{}}
        selectedParam={['takes', 'test1']}
        theme={mockTheme}
      />
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
