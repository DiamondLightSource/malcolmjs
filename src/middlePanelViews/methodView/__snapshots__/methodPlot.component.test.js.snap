// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MethodPlot renders correctly 1`] = `
<div
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
/>
`;
